src/main/cpp/src/NumberConverterJni.cpp

```diff
@@ -0,0 +1,248 @@
+  JNI_NULL_CHECK(env, from_base, "from_base column handle is null", 0);
+  JNI_NULL_CHECK(env, to_base, "to_base column handle is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto input_view{*reinterpret_cast<cudf::column_view const*>(input)};
+    auto from_base_view{*reinterpret_cast<cudf::column_view const*>(from_base)};
+    auto to_base_view{*reinterpret_cast<cudf::column_view const*>(to_base)};
+
+    return cudf::jni::release_as_jlong(
+      spark_rapids_jni::convert_cv_cv_cv(input_view, from_base_view, to_base_view));
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_NumberConverter_convertCvCvS(
+  JNIEnv* env, jclass, jlong input, jlong from_base, int to_base)
+{
+  JNI_NULL_CHECK(env, input, "input column handle is null", 0);
+  JNI_NULL_CHECK(env, from_base, "from_base column handle is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto input_view{*reinterpret_cast<cudf::column_view const*>(input)};
+    auto from_base_view{*reinterpret_cast<cudf::column_view const*>(from_base)};
+
+    return cudf::jni::release_as_jlong(
+      spark_rapids_jni::convert_cv_cv_s(input_view, from_base_view, to_base));
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_NumberConverter_convertCvSCv(
+  JNIEnv* env, jclass, jlong input, int from_base, jlong to_base)
+{
+  JNI_NULL_CHECK(env, input, "input column handle is null", 0);
+  JNI_NULL_CHECK(env, to_base, "to_base column handle is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto input_view{*reinterpret_cast<cudf::column_view const*>(input)};
+    auto to_base_view{*reinterpret_cast<cudf::column_view const*>(to_base)};
+
+    return cudf::jni::release_as_jlong(
+      spark_rapids_jni::convert_cv_s_cv(input_view, from_base, to_base_view));
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_NumberConverter_convertCvSS(
+  JNIEnv* env, jclass, jlong input, int from_base, int to_base)
+{
+  JNI_NULL_CHECK(env, input, "input column handle is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto input_view{*reinterpret_cast<cudf::column_view const*>(input)};
+
+    return cudf::jni::release_as_jlong(
+      spark_rapids_jni::convert_cv_s_s(input_view, from_base, to_base));
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_NumberConverter_convertSCvCv(
+  JNIEnv* env, jclass, jlong input, jlong from_base, jlong to_base)
+{
+  JNI_NULL_CHECK(env, input, "input string scalar is null", 0);
+  JNI_NULL_CHECK(env, from_base, "from_base column handle is null", 0);
+  JNI_NULL_CHECK(env, to_base, "to_base column handle is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto const& input_scalar = *reinterpret_cast<cudf::string_scalar*>(input);
+    auto from_base_view{*reinterpret_cast<cudf::column_view const*>(from_base)};
```

Comment:
`input_scalar` can be created by the same way as the other variables for consistency.